### ISSUE TITLE
CalendarDelegate.kt Compile Error with Flutter 3.7.6 • channel stable

### ISSUE THREAD

**Issue by steffomix** [builttoroam/device_calendar#475]
Created a new Flutter project with Null-Safety enabled with some dependencies as shown below and nothing else.

The dependencies:

  path: ^1.8.2
  http: ^0.13.5
  logger: ^1.1.0
  path_provider: ^2.0.11
  flutter_local_notifications: ^9.9.0
  permission_handler: ^10.0.0
  shared_preferences: ^2.0.15
  external_path: ^1.0.3
  android_intent_plus: ^3.1.6
  loading_animation_widget: ^1.2.0+4
  confirm_dialog: ^1.0.1
  background_location_tracker: ^1.4.0
  device_calendar: ^3.1.0
  app_settings: ^4.2.0

A "flutter run" created this output

e: /home/stefan/.pub-cache/hosted/pub.dev/device_calendar-3.1.0/android/src/main/kotlin/com/builttoroam/devicecalendar/CalendarDelegate.kt: (584, 73): Only safe (?.) or non-null asserted (!!.) calls are allowed on a nullable receiver of type Activity?
e: /home/stefan/.pub-cache/hosted/pub.dev/device_calendar-3.1.0/android/src/main/kotlin/com/builttoroam/devicecalendar/CalendarDelegate.kt: (585, 72): Only safe (?.) or non-null asserted (!!.) calls are allowed on a nullable receiver of type Activity?
e: /home/stefan/.pub-cache/hosted/pub.dev/device_calendar-3.1.0/android/src/main/kotlin/com/builttoroam/devicecalendar/CalendarDelegate.kt: (599, 36): Only safe (?.) or non-null asserted (!!.) calls are allowed on a nullable receiver of type Activity?
e: /home/stefan/.pub-cache/hosted/pub.dev/device_calendar-3.1.0/android/src/main/kotlin/com/builttoroam/devicecalendar/CalendarDelegate.kt: (680, 9): 'when' expression must be exhaustive, add necessary 'DAILY', 'HOURLY', 'MINUTELY', 'SECONDLY' branches or 'else' branch instead
e: /home/stefan/.pub-cache/hosted/pub.dev/device_calendar-3.1.0/android/src/main/kotlin/com/builttoroam/devicecalendar/CalendarDelegate.kt: (789, 65): Using 'max(): T?' is an error. Use maxOrNull instead.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':device_calendar:compileReleaseKotlin'.
> A failure occurred while executing org.jetbrains.kotlin.compilerRunner.GradleCompilerRunnerWithWorkers$GradleKotlinCompilerWorkAction
   > Compilation error. See log for more details

Is there any shorthand way to fix this or make a workaround?

**Comment by IVLIVS-III** [comment]
Please use `device_calendar: ^4.3.0` and report back if the issue persists. Depending on your Kotlin version, you could also try the latest pre-release.

**Comment by steffomix** [comment]
Thanks, that worked - nearly.
Still needed to add to ~/.pub-cache/hosted/pub.dev/device_calendar-4.3.0/android/src/main/kotlin/com/builttoroam/devicecalendar/CalendarDelegate.kt at line 789:

```
} else -> {
            	// do nothing
}
```
So that it all looks like that.

```
//
// ~/.pub-cache/hosted/pub.dev/device_calendar-4.3.0/android/src/main/kotlin/com/builttoroam/devicecalendar/CalendarDelegate.kt 
//
        when (rfcRecurrenceRule.freq) {
            Freq.WEEKLY, Freq.MONTHLY, Freq.YEARLY -> {
                recurrenceRule.daysOfWeek = rfcRecurrenceRule.byDayPart?.mapNotNull {
                    DayOfWeek.values().find { dayOfWeek -> dayOfWeek.ordinal == it.weekday.ordinal }
                }?.toMutableList()
            } else -> {
            	// do nothing
            }
        }
```

**May be it crashes during runtime but for at least I got it compiled.**

I also added disable gradle deprecatred warnings causing build failures
in app/build.gradle at the bottom of the file

```
//
// <flutter-project>/android/app/build.gradle
//
tasks.withType(JavaCompile) {
    options.compilerArgs << '-Xlint:-deprecation'
}
```

This is nothing I would recommend to anyone but just a quick fix

**Comment by IVLIVS-III** [comment]
@steffomix yep, great. And that's exactly the kotlin-error I thought might occur.
No source modifications should be necessary using Prerelease version [4.3.1-4217521123](https://pub.dev/packages/device_calendar/versions/4.3.1-4217521123) (except for maybe disabling gradle deprecated warnings).

**Comment by steffomix** [comment]
Thanks @IVLIVS-III, works like a charme now!
